ecommerce_extensions/payment/views/ednx_pp.py

```diff
@@ -0,0 +1,209 @@
+                        basket.id,
+                        ppr.id
+                    )
+                    return HttpResponse()
+        except:  # pylint: disable=bare-except
+            logger.exception('Attempts to handle payment for basket [%d] failed.', basket.id)
+            return HttpResponse(status=status.HTTP_200_OK)
+
+        try:
+            # Note (CCB): In the future, if we do end up shipping physical products, we will need to
+            # properly implement shipping methods. For more, see
+            # http://django-oscar.readthedocs.org/en/latest/howto/how_to_configure_shipping.html.
+            shipping_method = NoShippingRequired()
+            shipping_charge = shipping_method.calculate(basket)
+
+            # Note (CCB): This calculation assumes the payment processor has not sent a partial authorization,
+            # thus we use the amounts stored in the database rather than those received from the payment processor.
+            user = basket.owner
+            order_total = OrderTotalCalculator().calculate(basket, shipping_charge)
+            billing_address = self._get_billing_address(ednx_response)
+
+            self.handle_order_placement(
+                order_number,
+                user,
+                basket,
+                None,
+                shipping_method,
+                shipping_charge,
+                billing_address,
+                order_total,
+                request=request,
+            )
+
+            return HttpResponse()
+        except:  # pylint: disable=bare-except
+            payment_processor = self.payment_processor.NAME.title() if self.payment_processor else None
+            logger.exception(self.order_placement_failure_msg, payment_processor, basket.id)
+            return HttpResponse(status=status.HTTP_200_OK)
+
+    def get(self, request, *args, **kwargs):
+        # pylint: disable=unused-argument
+        """Handle an incoming user returned to us by PP after processing payment."""
+        ednx_response = request.GET.dict()
+        try:
+            transaction_id = ednx_response.get('transactionId')
+            order_number = ednx_response.get('referenceCode')
+            basket_id = OrderNumberGenerator().basket_id(order_number)
+
+            logger.info(
+                'Received PP payer notification for transaction [%s], associated with basket [%d].',
+                transaction_id,
+                basket_id,
+            )
+
+            basket = self._get_basket(basket_id)
+
+            if not basket:
+                logger.error('Received payer notification for non-existent basket [%s].', basket_id)
+                return redirect(reverse('payment_error'))
+        except:  # pylint: disable=bare-except
+            return redirect(reverse('payment_error'))
+
+        receipt_url = get_receipt_page_url(
+            order_number=basket.order_number,
+            site_configuration=basket.site.siteconfiguration,
+        )
+
+        try:
```

Comment:
this block doesn't make sense, since this catches the errors from `return redirect(receipt_url)` and then make the same  in the line 209 without any change